### PR TITLE
fix(streaming): fix topn cache for recovery

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Always initializes topn cache for topn executor in case of recovery.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/stream/src/executor/managed_state/top_n/top_n_state_new.rs [hunk 1]
```diff
@@ -175,6 +176,52 @@ impl<S: StateStore> ManagedTopNStateNew<S> {
...
+        epoch: u64,
+    ) -> StreamExecutorResult<()> {
+        assert!(topn_cache.low.is_empty());
+        assert!(topn_cache.middle.is_empty());
+        assert!(topn_cache.high.is_empty());
+
+        let state_table_iter = iter_state_table(&self.state_table, epoch, pk_prefix).await?;
+        pin_mut!(state_table_iter);
+        if topn_cache.offset > 0 {
+            while let Some(item) = state_table_iter.next().await {
```
Comment (StrikeW) [inline src/stream/src/executor/managed_state/top_n/top_n_state_new.rs:192]: Could you reuse the `fill_cache()` method to populate these caches(e.g. refactor the `start_key` to an `Option`)?  These repeated while loop is kind of verbose.

Comment (chenzl25) [reply src/stream/src/executor/managed_state/top_n/top_n_state_new.rs:192]: `init_topn_cache()` can populate these caches by calling `iter_state_table()` once, while reusing `fill_cache()` need three times.

Comment (StrikeW) [reply src/stream/src/executor/managed_state/top_n/top_n_state_new.rs:192]: Got it.

**Review by StrikeW — APPROVED** [review]
LGTM

**Pull request merged**